src/avram/bulk_upsert.cr

```diff
@@ -0,0 +1,82 @@
+class Avram::BulkUpsert(T)
+  def initialize(@records : Array(T), @column_names : Array(Symbol))
+    @records = set_timestamps(records)
+  end
+
+  def statement
+    <<-SQL
+    INSERT INTO #{table}(#{fields})
+      (select * from unnest(#{value_placeholders}))
+      ON CONFLICT (#{conflicts}) DO UPDATE SET #{updates}
+      RETURNING #{returning}
+    SQL
+  end
+
+  def args
+    @records.map do |record|
+      record.changed_attributes.map(&.value)
+    end.transpose
```

Comment:
If the records have a mismatching count of keys, transpose will fail.

Reply:
Yeah, we still need to add some sort of Way of preventing mismatched arrays, and handling the null overwrite issues in the failing spec

Reply:
Resolved! 🙇 compiler